File: 백준/Silver/5800. 성적 통계/성적 통계.py
import sys

k = int(sys.stdin.readline())
for x in range(1, k + 1):
    data = list(map(int, sys.stdin.readline().split()))
    scores = sorted(data[1:])
    max_diff = 0
    for i in range(data[0] - 1):
        max_diff = max(max_diff, scores[i + 1] - scores[i])
    print(f'Class {x}\nMax {scores[-1]}, Min {scores[0]}, Largest gap {max_diff}')

File: 백준/Silver/5800. 성적 통계/성적 통계.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int k;
	cin >> k;
	for (int i = 1; i <= k; i++) {
		int n;
		cin >> n;
		int scores[51];
		for (int j = 0; j < n; j++) {
			cin >> scores[j];
		}
		sort(scores, scores + n);
		int gap = 0;
		for (int j = 0; j < n - 1; j++) {
			int v = scores[j + 1] - scores[j];
			if (gap < v) gap = v;
		}
		cout << "Class " << i << '\n';
		cout << "Max " << scores[n - 1] << ", Min " << scores[0] << ", Largest gap " << gap << '\n';
	}
}